nullaway/src/main/java/com/uber/nullaway/fixserialization/FixLocation.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2022 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization;
+
+import com.google.common.base.Preconditions;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.tools.javac.code.Symbol;
+import java.net.URI;
+import javax.lang.model.element.ElementKind;
+
+/** Holds all the information to target a specific element in the source code. */
+public class FixLocation {
```

Comment:
This design is a bit convoluted.  How about splitting into:

* An interface `FixLocation`, with the methods invoked by other classes;
* An abstract class `AbstractFixLocation`, with shared fields and code among a different types of fix locations; and
* Three concrete classes extending `AbstractFixLocation`, one for each `Kind`.

I think with this design you can get rid of the `Kind` enum entirely; just have a `type` field in `AbstractFixLocation` and initialize it appropriately in each of the subclasses.

Reply:
bec0b95db437de05f0a21349aab4d0758e9ea794